modules/sfp_ipqualityscore.py

```diff
@@ -0,0 +1,171 @@
+    }
+
+    optdescs = {
+        "api_key": "IPQualityScore API Key",
+    }
+
+    errorState = False
+
+    def setup(self, sfc, userOpts=None):
+        if userOpts is None:
+            userOpts = {}
+        self.sf = sfc
+        self.results = self.tempStorage()
+        self.opts.update(userOpts)
+
+    def watchedEvents(self):
+        return [
+            "PHONE_NUMBER",
+            "EMAILADDR",
+            "IP_ADDRESS",
+            "DOMAIN_NAME"
+        ]
+
+    def producedEvents(self):
+        return [
+            "MALICIOUS_PHONE_NUMBER",
+            "MALICIOUS_EMAILADDR",
+            "MALICIOUS_IPADDR",
+            "MALICIOUS_INTERNET_NAME",
+            "RAW_RIR_DATA"
+        ]
+
+    def handle_error_response(self, qry, res):
+        try:
+            error_info = json.loads(res["content"])
+        except Exception:
+            error_info = None
+        if error_info:
+            error_message = error_info.get("message")
+        else:
+            error_message = None
+        if error_message:
+            error_str = f", message {error_message}"
+        else:
+            error_str = ""
+        self.sf.error(f"Failed to get results for {qry}, code {res['code']}{error_str}")
+
+    def query(self, qry, eventName):
+        queryString = ""
+        if eventName == "PHONE_NUMBER":
+            queryString = f"https://ipqualityscore.com/api/json/phone/{self.opts['api_key']}/{qry}"
+        elif eventName == "EMAILADDR":
+            queryString = f"https://ipqualityscore.com/api/json/email/{self.opts['api_key']}/{qry}"
+        elif eventName == "IP_ADDRESS" or eventName == "DOMAIN_NAME":
+            queryString = f"https://ipqualityscore.com/api/json/ip/{self.opts['api_key']}/{qry}"
+
+        res = self.sf.fetchUrl(
+            queryString,
+            timeout=self.opts["_fetchtimeout"],
+            useragent="SpiderFoot",
+        )
+
+        success = json.loads(res["content"]).get("success")
+        if res["code"] != "200" or not success:
+            self.handle_error_response(qry, res)
+            return None
+
+        if res['content'] is None:
+            self.sf.info(f"No IPQualityScore info found for {qry}")
+            return None
+
+        try:
+            return json.loads(res['content'])
+        except Exception as e:
+            self.sf.error(f"Error processing JSON response from IPQualityScore: {e}")
+
+        return None
+
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+
+        if self.errorState:
+            return None
+
+        self.sf.debug(f"Received event, {eventName}, from {srcModuleName}")
+
+        if self.opts["api_key"] == "":
+            self.sf.error(
+                f"You enabled {self.__class__.__name__} but did not set an API Key!"
+            )
+            self.errorState = True
+            return None
+
+        self.results[eventData] = True
```

Comment:
Even though you are storing this, you aren't doing anything with it. Was the intention to perform duplicate checking?